src/dune_rules/pkg_rules.ml

```diff
@@ -1175,13 +1189,40 @@ end = struct
           (Context_name.build_dir ctx)
           (Dune_pkg.Lock_dir.Pkg.files_dir info.name ~lock_dir)
       in
+      let* inherited_post_depends =
+        (* Deduplicate package names, preserving the association
+           between package names and (any) one of the locations of the
+           package name (for error messages). *)
+        let all_locs_by_name =
+          List.concat_map depends ~f:(fun (pkg : Pkg.t) ->
+            List.map pkg.post_depend_names ~f:(fun (loc, name) -> name, loc))
+          |> Package.Name.Map.of_list_reduce ~f:(fun x _ -> x)
+        in
+        Package.Name.Map.to_list all_locs_by_name
+        |> Memo.parallel_map ~f:(fun (name, loc) ->
+          if Package.Name.equal name info.name
+          then
+            (* It's possible for a package to be a post
```

Comment:
I don't really get this problem, is there a test for it?

Reply:
This is covered by the `transitive-post-deps` test. `foo` has a post dep on `bar`, and `bar` has a regular dep on `foo`. This loop is enumerating the post deps of each regular dep of a package, and `bar` is a post dep of `foo`, which is a regular dep of `bar`. Without this check, we'd try to resolve `bar` while computing the inherited post dependencies of `bar` which would be a memo loop.

Reply:
When you say "memo loop", I assume you mean a cycle during the _resolution_ of the packages (as opposed to a dependency cycle). If that is the case, then there's no need to introduce a workaround and we can just modify resolution not to cycle.